File: commonfamily/src/main/java/com/koenig/commonModel/Item.java
package com.koenig.commonModel;


import java.nio.ByteBuffer;
import java.util.UUID;

/**
 * Created by Thomas on 06.10.2017.
 */

public abstract class Item extends Byteable {
    protected String id;
    protected String name;

    public Item(String name) {
        this.name = name;
        id = UUID.randomUUID().toString();
    }

    public Item(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public Item(ByteBuffer buffer) {
        id = byteToString(buffer);
        name = byteToString(buffer);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    @Override
    public int getByteLength() {
        return getStringLength(id) + getStringLength(name);
    }

    @Override
    public void writeBytes(ByteBuffer buffer) {
        buffer.put(stringToByte(id));
        writeString(name, buffer);
    }

}

File: commonfamily/src/main/java/com/koenig/commonModel/Item.kt
package com.koenig.commonModel


import java.nio.ByteBuffer
import java.util.*

/**
 * Created by Thomas on 06.10.2017.
 */

abstract class Item : Byteable {
    var id: String
    var name: String

    override val byteLength: Int
        get() = Byteable.Companion.getStringLength(id) + Byteable.Companion.getStringLength(name)

    constructor(name: String) {
        this.name = name
        id = UUID.randomUUID().toString()
    }

    constructor(id: String, name: String) {
        this.id = id
        this.name = name
    }

    constructor(buffer: ByteBuffer) {
        id = Byteable.byteToString(buffer)
        name = Byteable.byteToString(buffer)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val item = other as Item?

        return id == item!!.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun writeBytes(buffer: ByteBuffer) {
        buffer.put(Byteable.Companion.stringToByte(id))
        Byteable.Companion.writeString(name, buffer)
    }

}
